consensus/mint-client/src/config.rs

```diff
@@ -44,29 +196,30 @@ pub enum Commands {
         #[clap(long, env = "MC_CONSENSUS_URI")]
         node: ConsensusClientUri,
 
-        /// The key to sign the transaction with.
-        #[clap(long, parse(try_from_str = load_key_from_pem), env = "MC_MINTING_SIGNING_KEY")]
-        signing_key: Ed25519Private,
-
-        /// The b58 address we are minting to.
-        #[clap(long, parse(try_from_str = parse_public_address), env = "MC_MINTING_RECIPIENT")]
-        recipient: PublicAddress,
+        #[clap(flatten)]
+        params: MintTxParams,
+    },
 
-        /// The token id we are minting.
-        #[clap(long, env = "MC_MINTING_TOKEN_ID")]
-        token_id: u32,
+    // Generate a MintTx and write it to a JSON file.
+    GenerateMintTx {
+        /// Filename to write the mint configuration to.
+        #[clap(long, env = "MC_MINTING_OUT_FILE")]
+        out: PathBuf,
 
-        /// The amount we are minting.
-        #[clap(long, env = "MC_MINTING_AMOUNT")]
-        amount: u64,
+        #[clap(flatten)]
+        params: MintTxParams,
+    },
 
-        /// Tombstone block.
-        #[clap(long, env = "MC_MINTING_TOMBSTONE")]
-        tombstone: Option<u64>,
+    // Submit json-encoded MintTx(s). If multiple transactions are provided, signatures will
+    // be merged.
+    SubmitMintTx {
+        /// URI of consensus node to connect to.
+        #[clap(long, env = "MC_CONSENSUS_URI")]
+        node: ConsensusClientUri,
 
-        /// Nonce.
-        #[clap(long, parse(try_from_str = FromHex::from_hex), env = "MC_MINTING_NONCE")]
-        nonce: Option<[u8; NONCE_LENGTH]>,
+        /// Filename to read the mint configuration from.
```

Comment:
Seems useful to specify the format
```suggestion
        /// Paths for the JSON-formatted mint configuration files.
```

Reply:
Its whatever serde comes up with and is not meant to be consumed by humans or any other tool.

Reply:
I can mentioned what kind of object gets serialized there